openedx/core/djangoapps/user_authn/views/utils.py

```diff
@@ -108,3 +115,56 @@ def get_mfe_context(request, redirect_to, tpa_hint=None):
         'countryCode': country_code,
     })
     return context
+
+
+def _get_username_prefix(data):
+    """
+    Get the username prefix (name initials) based on the provided data.
+
+    Args:
+    - data (dict):  Registration payload.
+
+    Returns:
+    - str: Name initials or None.
+    """
+    username_regex_partial = settings.USERNAME_REGEX_PARTIAL
+    full_name = ''
+    if data.get('first_name', '').strip() and data.get('last_name', '').strip():
+        full_name = f"{unidecode(data.get('first_name', ''))} {unidecode(data.get('last_name', ''))}"
+    elif data.get('name', '').strip():
+        full_name = unidecode(data['name'])
+
+    if full_name.strip():
+        full_name = re.findall(username_regex_partial, full_name)[0]
+        name_initials = "".join([name_part[0] for name_part in full_name.split()[:2]])
+        return name_initials.upper() if name_initials else None
+
+    return None
+
+
+def get_auto_generated_username(data):
+    """
+    Generate username based on learner's name initials, current date and configurable random string.
+
+    This function creates a username in the format <name_initials>_<YYMM>_<configurable_random_string>
+
+    The length of random string is determined by AUTO_GENERATED_USERNAME_RANDOM_STRING_LENGTH setting.
+
+     Args:
+    - data (dict):  Registration payload.
+
+    Returns:
+    - str: username.
+    """
+
+    # We generate the username regardless of whether the name is empty or invalid. We do this
+    # because the name validations occur later, ensuring that users cannot create an account without a valid name.
```

Comment:
Can you move this either in the docstring or above the return statement at the end?